File: ProjetoAtleta/AtletaJava/Atleta.java
package AtletaJava;

import java.util.Objects;

public class Atleta {
	String nome;
	String fone;
	String apelido;
	String dataNasc;
	int pontuacaoAcumulada;
	
	public Atleta(String nome, String fone, String apelido, String data, int pontos){
		this.nome = nome;
		this.fone = fone;
		this.apelido = apelido;
		this.dataNasc = data;
		this.pontuacaoAcumulada = pontos;
	}
	
	public Atleta(){
		
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getFone() {
		return fone;
	}

	public void setFone(String fone) {
		this.fone = fone;
	}

	public String getApelido() {
		return apelido;
	}

	public void setApelido(String apelido) {
		this.apelido = apelido;
	}

	public String getDataNasc() {
		return dataNasc;
	}

	public void setDataNasc(String dataNasc) {
		this.dataNasc = dataNasc;
	}

	public int getPontuacaoAcumulada() {
		return pontuacaoAcumulada;
	}

	public void setPontuacaoAcumulada(int pontuacaoAcumulada) {
		this.pontuacaoAcumulada = pontuacaoAcumulada;
	}

	@Override
	public String toString() {
		return "Atleta [nome=" + nome + ", fone=" + fone + ", apelido=" + apelido + ", dataNasc=" + dataNasc
				+ ", pontuacaoAcumulada=" + pontuacaoAcumulada + "]";
	}

	@Override
	public int hashCode() {
		return Objects.hash(fone);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Atleta other = (Atleta) obj;
		return Objects.equals(fone, other.fone);
	}

	
	
	
}

File: ProjetoAtleta/AtletaPython/Atleta.py
class Atleta:
    def __init__(self, nome, apelido, fone, data_nascimento, pontuacao):
        self.nome = nome
        self.apelido = apelido
        self.fone = fone
        self.data_nascimento = data_nascimento
        self.pontuacao = pontuacao


    def get_nome(self):
        return self.nome

    def set_nome(self, nome):
        self.nome = nome

    def get_apelido(self):
        return self.apelido

    def set_apelido(self, apelido):
        self.apelido = apelido

    def get_fone(self):
        return self.fone

    def set_fone(self, fone):
        self.fone = fone

    def get_data_nascimento(self):
        return self.data_nascimento

    def set_data_nascimento(self, data_nascimento):
        self.data_nascimento = data_nascimento

    def get_pontuacao(self):
        return self.pontuacao

    def set_pontuacao(self, pontuacao):
        self.pontuacao = pontuacao

    def __eq__(self, other):
        if isinstance(other, Atleta):
            return self.fone == other.fone
   
        return False
    def __str__(self):
        return f"Atleta: {self.nome}, Apelido: {self.apelido}, Fone: {self.fone}, Data de Nascimento: {self.data_nascimento}, Pontuação: {self.pontuacao}"
   
    def __hash__(self):
        return hash(self.fone)